src/Plugin/Condition/DataIsEmpty.php

```diff
@@ -36,17 +36,10 @@ class DataIsEmpty extends RulesConditionBase {
   /**
    * {@inheritdoc}
    */
-  public function evaluate() {
-    $data = $this->getContext('data')->getContextData();
+  protected function doEvaluate($data) {
     if ($data instanceof ComplexDataInterface || $data instanceof ListInterface) {
       return $data->isEmpty();
```

Comment:
So this is now dead code and will never be executed, right? $data will never conform to those interfaces because $data is not typed data here?